### PR TITLE
fix: https://github.com/aporeto-inc/aporeto-devs/issues/820

### PR DESCRIPTION
**Pull request by aaslamin**
Addresses part of https://github.com/aporeto-inc/aporeto-devs/issues/820 by adding a new field `EnforcerID` to represent the ID of the Enforcer. Maintains backwards compatibility by keeping the `ID` field, but now using it with the proper attributes required for persistence by the Control Plane.

---

_FYI @0xmchadha this will come handy for your corresponding enforcer PR_

### CODE REVIEW

File: doc/documentation.md [hunk 1]
```diff
@@ -3587,7 +3587,13 @@ Type: `float`
...
 Total CPU utilization of the Defender as a percentage of vCPUs.
 
-##### `ID` [`required`]
+##### `ID` [`identifier`,`autogenerated`,`read_only`]
+
+Type: `string`
+
+Identifier of the object.
+
+##### `enforcerID` [`required`]
```
Comment (primalmotion) [inline doc/documentation.md:3596]: sadly if you mark this as required, it won't reach your code. We have to remove the fact that it is required for now, and we must do a custom entity validation ensuring everything is correct

**Review by primalmotion — DISMISSED** [review]

File: doc/documentation.md (hunk 1 above)
Comment (aaslamin) [reply doc/documentation.md]: @primalmotion great catch 🏆 

I added a custom entity validation and removed the check from the backend. Now the backend processor only needs to deal with the new `EnforcerID` field.

File: custom_validations_test.go [hunk 2]
```diff
@@ -1143,6 +1143,62 @@ AiEA0epxATHNzheAa8ZuiPeNQL6DhoKYz3B+41J2vgVlGZY=
...
+			// dealing with the 'ID' field being set
+			if err == nil {
+				if scenario.report.ID != "" {
+					t.Errorf("Validation passed, but the report's 'ID' field was NOT reset. It contained the value: %s", scenario.report.ID)
+				}
+
+				if scenario.report.EnforcerID == "" {
+					t.Error("Validation passed, but the report's 'enforcerID' field was empty")
+				}
+			}
```
Comment (aaslamin) [inline custom_validations_test.go:1197]: 👀 @primalmotion

Comment (t00f) [reply custom_validations_test.go:1197]: 👍

**Review by t00f — APPROVED** [review]
LGTM

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerreport-deprecate-id-field",
    "component": "gaia",
    "pr-id": "546",
    "commit-sha": "f0a216e48bde3d14a2e3e4379de5a5fd4dc1c1de",
    "pipeline": "master"
  },
  {
    "project": "enforcerreport-deprecate-id-field",
    "component": "backend",
    "pr-id": "852",
    "commit-sha": "527e6356cb39b1f69d3ddf29cd1449544f8f0bc7",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerreport-deprecate-id-field",
    "component": "gaia",
    "pr-id": "546",
    "commit-sha": "f0a216e48bde3d14a2e3e4379de5a5fd4dc1c1de",
    "pipeline": "master"
  },
  {
    "project": "enforcerreport-deprecate-id-field",
    "component": "backend",
    "pr-id": "852",
    "commit-sha": "527e6356cb39b1f69d3ddf29cd1449544f8f0bc7",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerreport-deprecate-id-field",
    "component": "gaia",
    "pr-id": "546",
    "commit-sha": "ecc1e43630b7735be6c13333259bbe0689924558",
    "pipeline": "master"
  },
  {
    "project": "enforcerreport-deprecate-id-field",
    "component": "backend",
    "pr-id": "852",
    "commit-sha": "157dcd462955b76ac23363141e27bb9142d4f32e",
    "pipeline": "master"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "enforcerreport-deprecate-id-field",
    "component": "gaia",
    "pr-id": "546",
    "commit-sha": "ecc1e43630b7735be6c13333259bbe0689924558",
    "pipeline": "master"
  },
  {
    "project": "enforcerreport-deprecate-id-field",
    "component": "backend",
    "pr-id": "852",
    "commit-sha": "157dcd462955b76ac23363141e27bb9142d4f32e",
    "pipeline": "master"
  }
]
```

**Pull request merged**